assets/js/hocs/with-searched-products.js

```diff
@@ -54,6 +54,7 @@ const withSearchedProducts = createHigherOrderComponent(
 
 			onSearch( search ) {
 				const { selected } = this.props;
+				this.setState( { loading: true } );
```

Comment:
Is this `setState()` is required considering we are already setting `loading: true` in line 83?

Reply:
Line 83 doesn't run unless defer is enabled right?

Reply:
Right, but `onSearch` only runs when defer is enabled too. Notice when `IS_LARGE_CATALOG` is false we pass `null` to `onSearch`.

In small catalogs, search is done client-side without making any API request.

Reply:
O maybe there is another bug here then! Large catalog is enabled after 200, but there is a hardcoded 100 limit on requests.

Reply:
Good point! Is there any reason for it being 200 instead of 100? I see it was introduced in #515 but I'm not sure to understand why 200 instead of 100.

Reply:
I agree with @Aljullu that there's no need for two calls to `this.setState( { loading: true } )` here.  As far as I can tell you just need to invoke it in the anonymous callback on the `onSearch` prop.  The `onSearch` callback isn't called anywhere else (other than debouncedSearch - which is already covered by the callback on the `onSearch` prop).

Reply:
> As far as I can tell you just need to invoke it in the anonymous callback on the onSearch prop.

Considering search is debounced, I think it's better to have `setState( { loading: true } )` right before the `debouncedOnSearch` call (line 82) than inside `onSearch` (line 57). Otherwise for some milliseconds the loading spinner would not be shown.

Reply:
Sorted 👍